src/platform/testing/visual-regression/index.js

```diff
@@ -1,100 +1,124 @@
+      console.log(chalk.red(`Couldn't screenshot ${route}`));
+    }
+    /* eslint-enable */
+    // await page.goto('about:blank')
+  }
+  table.push(...results.map(tableFormatter));
+  /* eslint-disable  no-console */
+  console.log(table.toString());
+  /* eslint-enable */
 }
 
 // A wrapper around the login helper to return a promise
-function login(browser) {
+async function login(page) {
   const token = Auth.getUserToken();
-
-  return new Promise(resolve => {
-    Auth.logIn(token, browser, '/', 1);
-    browser.waitForElementVisible('body', Timeouts.normal, resolve);
-  });
-}
-
-// A wrapper around a helper function for grabbing the sitemap.xml and converting into an array of URL's
-function getSiteRoutes() {
-  return new Promise(resolve => sitemapURLs(resolve));
+  await Auth.logIn(token, page, '/', 1);
+  await page.waitForSelector('body', { timeout: Timeouts.normal });
 }
 
 // Checks command flags to determine the "routeHandler" function that will perform a task after the browser navigates to each route.
 function getRouteHandler() {
-  const commands = {
-    CREATE_BASELINE_IMAGES: 'baseline',
-    CALCULATE_DIFFS: 'diff'
-  };
-
-  const { command } = commandLineArgs([
-    { name: 'command', type: String },
-    { name: 'config', type: String, alias: 'c' }
-  ]);
-
   switch (command) {
     case commands.CREATE_BASELINE_IMAGES:
       return createBaselineImage;
-
     case commands.CALCULATE_DIFFS:
     default:
       return calculateDiff;
   }
 }
 
 // Returns a function that will be executed as a Nightwatch test case.
-function beginApplication(browser) {
+async function beginApplication(browser) {
   const routeHandler = getRouteHandler();
-
-  // Tests are async, so "done" is used as a callback to Nightwatch once we're finished.
-  browser.perform(done => {
-
-    // Before starting any routes, log the user in to prevent load errors.
-    login(browser)
-
-      // Parse the sitemap XML file into an array of URL's
-      .then(getSiteRoutes)
-
-      // Create a single long-running promise out of the routes array
-      .then(routes => createRouteHandlerChain(browser, routes, routeHandler))
-
-      // Close the browser window so Electron instances don't pile up.
-      .then(() => browser.closeWindow())
-
-      // Tell Nighwatch we're finished.
-      .then(done);
-  });
+  // Before starting any routes, log the user in to prevent load errors.
+  const page = await browser.newPage();
+  if (mobile === commands.MOBILE) {
+    page.emulate(iPhone);
+  }
+  await login(page);
+  // Parse the sitemap XML file into an array of URL's
+  const sitemap = await sitemapURLs();
+  const routes = sitemap.urls;
+  await createRouteHandlerChain(page, routes, routeHandler);
+  await page.close();
 }
 
 // The entry point for everything.
-function setup() {
-  return createE2eTest(beginApplication);
+async function setup() {
+  // check if mobile test or not
+  const browser = await puppeteer.launch({ headless: false });
```

Comment:
Does this work headless? I would assume so, so maybe we could make that the default? Given that it takes a while, it might be easier to use that way.

Reply:
It's beautifully quick with `headless: true` and zero bugs so far, runs through the whole site on desktop and mobile in 15min